virtual_rainforest/models/litter/litter_model.py

```diff
@@ -87,16 +110,40 @@ def __init__(
     ):
         super().__init__(data, update_interval, **kwargs)
 
-        # Check that litter pool data is appropriately bounded
-        if (
-            np.any(data["litter_pool_above_metabolic"] < 0.0)
-            or np.any(data["litter_pool_above_structural"] < 0.0)
-            or np.any(data["litter_pool_woody"] < 0.0)
-            or np.any(data["litter_pool_below_metabolic"] < 0.0)
-            or np.any(data["litter_pool_below_structural"] < 0.0)
-        ):
+        # Check that no litter pool is negative
+        all_pools = [
+            "litter_pool_above_metabolic",
+            "litter_pool_above_structural",
+            "litter_pool_woody",
+            "litter_pool_below_metabolic",
+            "litter_pool_below_structural",
+        ]
+        negative_pools = []
+        for pool in all_pools:
+            if np.any(data[pool] < 0):
+                negative_pools.append(pool)
+
+        if negative_pools:
             to_raise = InitialisationError(
-                "Initial litter pools contain at least one negative value!"
+                f"Pool sizes in {negative_pools} must be greater than 0!"
+            )
+            LOGGER.error(to_raise)
+            raise to_raise
+
+        # Check that lignin proportions are between 0 and 1
+        lignin_proportions = [
+            "lignin_above_structural",
+            "lignin_woody",
+            "lignin_below_structural",
+        ]
+        bad_proportions = []
+        for lignin_prop in lignin_proportions:
+            if np.any(data[lignin_prop] < 0) or np.any(data[lignin_prop] > 1):
```

Comment:
Even if it's just something that gets run at model initialisation for now I think it would be useful, as most variables I've used do have hard limits

Reply:
I forgot - we have already had a discussion about this in model init! I guess this is one of those cases where we might be implementing something that defends against errors at a low level but it still feels appropriate that models sanitize inputs.

Reply:
I think the question of "do we want to _immediately_ kill the simulation if a clear invalid value emerges" is a thorny one, whereas "do we want to stop the simulation from starting with invalid values" is easy to answer (we do).

Reply:
It's thorny in that it's something that we probably want eventually, but could get annoying if implemented prematurely

Reply:
That's a really good point - checking for initial values takes the load off `__init__` and means that shared variables are checked once, but emergent bad values still need to be trapped. I guess one thing to do is to have a routine that logs the bounds issue and sets the values to `np.nan`. That then allow the simulation to continue (possibly not for very long).

Reply:
The bounds checking was assigned to me a long time ago but I wonder if that would be better implemented by RCS? It would definitely be useful as we all reached that point where we have some internal hacky version of it?